bootstrap/terraform/vm.tf

```diff
@@ -0,0 +1,216 @@
+terraform {
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "3.29.1"
+    }
+    azuread = {
+      source  = "hashicorp/azuread"
+      version = "2.30.0"
+    }
+    http = {
+      source  = "hashicorp/http"
+      version = "3.2.0"
+    }
+  }
+}
+
+data "azurerm_client_config" "current" {}
+provider "azuread" {}
+provider "azurerm" {
+  features {}
+}
+
+resource "azuread_application" "vault_azure_app" {
+  display_name = "vault_azure_app"
+}
```

Comment:
Not a big deal as it is a small local setup config but if you ever work on larger production Terraform projects it is recommended to [not repeat the resource type in it's name](https://www.terraform-best-practices.com/naming). Like "...application" & "...app" & other examples below.

It is also recommended by HashiCorp to [prefix the resource name with the provider's name](https://developer.hashicorp.com/terraform/plugin/sdkv2/best-practices/naming).

And a personal preference of mine is to also have a "namespace" prefix to group resource that depend on one another. And name the resource based on it's purpose in the overall project.

Given all this resource names could be:
`vault_azure_tests`
With an extra section if you need disambiguation between resources of the same type, for example with assignments:
```
vault_azure_tests_system
vault_azure_tests_user
```

But yeah naming is always a topic with tons of grey area so I don't have a strong opinion, just some pointers from docs.

Reply:
Nice! Thanks for this info. I was unsure of what to do with these names. I'll change them 👍

Reply:
Done in [c3e62ec](https://github.com/hashicorp/vault-plugin-auth-azure/pull/65/commits/c3e62ec7d57e031fa73f5f262576bb55d5d45c99).